### PR TITLE
magento/magento2-login-as-customer#161: Message about choosing options for item is shown on "Thank you for your purchase!" page.

### PR DESCRIPTION
**Pull request by engcom-Foxtrot**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/29115: Message about choosing options for item is shown on "Thank you for your purchase!" page.

### Manual testing scenarios (*)

1. See https://github.com/magento/magento2/issues/29115: Message about choosing options for item is shown on "Thank you for your purchase!" page.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked cases with Disable Page Cache For Admin User=Yes/No
Result:
#178PRAdminASCsutomer

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked cases with Disable Page Cache For Admin User=Yes/No with latest [commit](https://github.com/magento/magento2-login-as-customer/commit/15134529b275d5107c0c4102703a6d0d2824153a)
Result:
#178PRAdminASCsutomer

**Notice:** Issue #172 is not related to this PR

File: app/code/Magento/Theme/Controller/Result/MessagePlugin.php [hunk 1]
```diff
@@ -88,12 +96,13 @@ public function __construct(
...
      * @param ResultInterface $result
      * @return ResultInterface
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function afterRenderResult(
         ResultInterface $subject,
         ResultInterface $result
     ) {
-        if (!($subject instanceof Json)) {
+        if (!$this->request->isAjax()) {
```
Comment (naydav) [inline app/code/Magento/Theme/Controller/Result/MessagePlugin.php:105]: Should keep BC
`!($subject instanceof Json) && !$this->request->isAjax()`

Comment (engcom-Foxtrot) [reply app/code/Magento/Theme/Controller/Result/MessagePlugin.php]: @naydav updated

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked cases with Disable Page Cache For Admin User=Yes/No
**Result:**
#178LACPR

**Comment by naydav** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
This fix is incorrect.

**Pull request closed without merging**